### DISCUSSION TITLE
有关迷你模式、界面UI和文件图标设计方面的设想

### DISCUSSION THREAD

**Discussion by TaicEart** [zhongyang219/MusicPlayer2#435] [Ideas]
从2009年用千千静听5.5.1（由于后续“音乐窗”这种影响听音乐的设计，不再更新了）到现在，一直在期待能再有一个代替这11年前的本地音乐播放器。国内外大多本地音乐播放器我都有试过，中间我还用过[AirPlay1~3](https://baike.baidu.com/item/AirPlay/6180018)（记得其中一个版本有惊艳的右键象限菜单），但该软件如今连官网也不存在了。

我使用了一段时间，也将9页开启的issues都翻了一遍，有许多想法，请让我分几个帖来叙述。

### 迷你模式
首先看了wiki，发现[迷你模式](https://github.com/zhongyang219/MusicPlayer2/wiki/%E8%BF%B7%E4%BD%A0%E6%A8%A1%E5%BC%8F)不能定制。当前迷你模式过于追求全了，但其实有桌面歌词的情况下歌词是不必要的，如此小的空间也不宜再划一块放频谱——我建议把频谱作为迷你模式的动态背景，或者允许用户直接不要。
若能自行定制，我们可以自己解决。

### 界面UI
主界面上迷你模式的三形按钮容易与汉堡包菜单（又称“抽屉菜单”按钮）混淆，这期间我多次把迷你模式按钮当作菜单点下。这个按钮推荐做成◳形或↗形。
同样的，迷你模式界面上“显示/隐藏播放列表”按钮也是三形，这非常容易误触……（还原主界面按钮推荐做成◱或↙）
UI上按钮若能自行定制，我可以自行设计来解决。

虽然使用了较长时间，但还是记不住界面3、界面4和界面5……有什么区别，如果这些界面允许在菜单后方备注区别，可以更方便选择。

播放列表显示方式应允许像千千静听那样自定义/自适应。同人音声的音频经常有艺术家或标题不全的情况，尤其是无损wav。可设计为如果遇到空缺则忽略该项，或直接显示文件名，或其它算法……
当然若能像千千静听那样可自定义，这个也可以自行解决。

### 关联文件图标
这就是我本行了。我多年前发布过一套文件图标 https://github.com/TaicEart/Undrgrnd ，每个文件图标都遵循其常见的“印象色”，而不是一类文件清一色，但同类文件又画风一致，比如flv是flashplayer那样的红色，webm是官方图标的绿色，虽然是互补色，但这2个文件图标都能被认知为视频流。
其中音乐类文件的wav图标如图，每个图标都支持Windows主要的4个[离散大小](https://learn.microsoft.com/zh-cn/windows/win32/shell/thumbnail-providers)且对齐像素：
![Preview.png](https://github.com/TaicEart/Undrgrnd/raw/master/Preview.png)
如果觉得可以，我可以免费提供，并追加——最近还做了opus的文件图标。如果觉得有需要修改之处也可以告诉我。

其实我还希望可视化频谱可以做成上图的样子，作为软件的动态背景🥰

其它设想和问题另发。

追加：MusicPlayer2.exe主程序中包含的144个图标，其中许多是直接缩小，未对齐像素导致模糊，第8~13的小尺寸图标模糊得尤其明显，这些需要修正。

**Comment by zhongyang219** [comment]
感谢你的建议。目前部分图标已经做了一些调整，接下来打算考虑添加迷你模式自定义的功能。
另外“界面3”、“界面4”这些名字是可以改的，可以到程序所在目录下的skins目录中打开对应的xml文件，在root节点中加上name属性，例如：`name="简洁UI"`，保存后重新启动MusicPlayer2即可。

**Reply by TaicEart** [reply]
报错，又发现一个图标问题：playlist（`\MusicPlayer2.exe,66`）关联图标的256尺寸看上去损坏了，资源管理器中最大只能显示48尺寸。
